### PR TITLE
feat(account-deletion): adding in account deletion options

### PR DESCRIPTION
**Pull request by bassrock**
## Summary

Creates the basic scaffolding for:
* Account Management Screen
* Delete Account Screen
* Button Styles

## References 

- IN-773

## Implementation Details
* I took time to build out Pocket Button styles within the Textile class including pressed states which we did not have.
* I had to implement a custom toggle style for the checkboxes that is in Textile
* I also worked to ensure we could start to use swift previews by following the [opinion piece here](https://medium.com/swift2go/swiftui-opinion-viewmodel-doesnt-belong-in-previews-62d9e1485b38)

## Test Steps

- [ ] Given I am a user
 And I click settings
 I can see Account Management
 
- [ ] Given I am a user
 And I click settings
And I click Account Management
I can see Delete Account

- [ ] Given I am a user
 And I click settings
And I click Account Management
And I click on Delete Account
I can see a delete confirmation view

- [ ] Given I am a free user
 And navigate to delete confirmation view
I see a check box to confirm I know it will delete my account

- [ ] Given I am a premium user
 And navigate to delete the confirmation view
I see a check box to confirm I know it will delete my account
and I see a checkbox to confirm I have canceled premium

- [ ] Given I am a user
 And navigate to delete the confirmation view
And I have clicked the checkboxes
The delete account button becomes clickable

- [ ] Given I am a premium user
 And navigate to delete the confirmation view
And I have clicked the I have canceled premium checkbox
The deleting pocket help button disappears

- [ ] Given I am a premium user
 And navigate to delete the confirmation view
And I click the cancelling premium help button
I am taken to a help article within pocket on how to cancel premium

## PR Checklist:
- [x] Added Unit / UI tests
- [x] Self Review (review, clean up, documentation, run tests)
- [x] Basic Self QA

## Screenshots

https://user-images.githubusercontent.com/1010384/222850497-9c60378f-bbd4-44fd-97f4-54ff00b63e04.mov

https://user-images.githubusercontent.com/1010384/222850530-6e920a15-ae70-488d-a60d-cc15fd595d5d.mov

### CODE REVIEW

**Review by Gio2018 — COMMENTED** [review]
This looks good @bassrock !

Just found a couple of UI issues (see screenshots): 
1. in dark mode, the text associated with the checkbox and the cancel button are not well visible, 
2. on iPad, the "Account Management" Header is truncated in the master view
We can either fix those here or in a follow up PR, up to you

Delete Account Form | Account Management Header
-- | --
 |

File: PocketKit/Sources/PocketKit/Settings/DeleteAccountView.swift [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+                    .padding(.top, 50)
+                    .padding()
+
+                Text(L10n.Settings.AccountManagement.DeleteAccount.warning)
+                    .style(.deleteAccountView.warning)
+                    .frame(maxWidth: .infinity, alignment: .leading)
+                    .padding()
+
+                /**
+                 Note below we use LocalizedStringKey. This is a total hack. This is because a SwiftUI text object treats Text("testing") differently then Text(someText). When passed as a variable, the Text view performs no localization, but it also does not render the embedded markdown. When cast to a LocalizedStringKey the Text field will try and localize the content (and fail) but it will render the markdown and bold the text.
```
Comment (bassrock) [inline PocketKit/Sources/PocketKit/Settings/DeleteAccountView.swift:41]: This was fun.. thanks @JohnOberhauser for the 🦆

**Comment by bassrock** [comment]
pushed swiftlint fix.

File: PocketKit/Sources/PocketKit/Report/ReportRecommendationView.swift [hunk 2]
```diff
@@ -73,17 +70,9 @@ struct ReportRecommendationView: View {
...
-                            .style(Style.submitButtonStyle)
-                        Spacer()
-                    }
-                }
-                .frame(height: Constants.submitButtonHeight)
-                .tint(Constants.submitButtonTintColor)
-                .background(Constants.submitButtonBackgroundColor)
-                .cornerRadius(Constants.cornerRadius)
+                    Text(isReported ? L10n.reported : L10n.submitFeedback)
+                }.buttonStyle(PocketButtonStyle(.primary))
```
Comment (cyndichin) [inline PocketKit/Sources/PocketKit/Report/ReportRecommendationView.swift:74]: we have an `ActionsPrimaryButtonStyle` that prob can be replaced, but out of scope for this

File: PocketKit/Sources/PocketKit/Settings/DeleteAccountView.swift [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+    @Environment(\.dismiss)
+    var dismiss
+
+    var body: some View {
+        NavigationView {
+            VStack(spacing: 0) {
+                Text(L10n.Settings.AccountManagement.deleteYourAccount)
+                    .style(.deleteAccountView.header)
+                    .padding(.top, 50)
```
Comment (cyndichin) [inline PocketKit/Sources/PocketKit/Settings/DeleteAccountView.swift:32]: not sure if we can across a consensus, but we discussed about possible putting magic numbers in an `enum`
https://pocket.slack.com/archives/C01UH57EJKD/p1677173129600309

**Review by cyndichin — APPROVED** [review]
looks great! love the enhancement and previews. Some comments but none blockers

File: PocketKit/Sources/PocketKit/Settings/DeleteAccountView.swift (hunk 1 above)
Comment (bassrock) [reply PocketKit/Sources/PocketKit/Settings/DeleteAccountView.swift:41]: @cyndichin nope, its cause of how SwiftUI worked. Remind me to show you in office hours. its the most annoying thing.

File: PocketKit/Sources/PocketKit/Settings/DeleteAccountView.swift (hunk 3 above)
Comment (bassrock) [reply PocketKit/Sources/PocketKit/Settings/DeleteAccountView.swift:32]: Making a note to come back to this.

**Pull request merged**